routingapi/api.go

```diff
@@ -0,0 +1,145 @@
+package routingapi
+
+import (
+	"fmt"
+	"time"
+
+	"code.cloudfoundry.org/route-registrar/config"
+
+	uaaclient "code.cloudfoundry.org/uaa-go-client"
+	uaaconfig "code.cloudfoundry.org/uaa-go-client/config"
+
+	"code.cloudfoundry.org/routing-api/models"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/routing-api"
+)
+
+type RoutingAPI interface {
+	Init(api_config config.RoutingAPI) error
+	RegisterRoute(route config.Route) error
+	UnregisterRoute(route config.Route) error
+	Close()
+}
+
+type apiState struct {
+	logger          lager.Logger
+	uaaClient       uaaclient.Client
+	apiClient       routing_api.Client
+	routerGroupGUID map[string]string
+}
+
+func NewRoutingAPI(logger lager.Logger) RoutingAPI {
+	return &apiState{
+		logger:          logger,
+		routerGroupGUID: make(map[string]string),
+	}
+}
+
+func buildOAuthConfig(config config.RoutingAPI) *uaaconfig.Config {
+
+	return &uaaconfig.Config{
+		UaaEndpoint:           config.OAuthURL,
+		SkipVerification:      config.SkipSSLValidation,
+		ClientName:            config.ClientID,
+		ClientSecret:          config.ClientSecret,
+		MaxNumberOfRetries:    3,
+		RetryInterval:         500 * time.Millisecond,
+		ExpirationBufferInSec: 30,
+		CACerts:               config.CACerts,
+	}
+}
+
+func (a *apiState) Init(config config.RoutingAPI) error {
```

Comment:
Can we please get rid of this init and just inject a `routingAPI` client into `NewRoutingAPI`? Would like to not mutate `apiState` after its been newed up.

All of this should be setup in `main.go` if possible.